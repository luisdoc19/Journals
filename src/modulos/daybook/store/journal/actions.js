import journalApi from "@/api/journalApi";

export const loadEntries = async ({ commit }) => {
  const { data } = await journalApi.get("/entries.json");
  if (!data) {
    commit("setEntries", []);
    return;
  }

  const entries = [];
  for (let id of Object.keys(data)) {
    entries.push({
      id,
      ...data[id],
    });
  }
  commit("setEntries", entries);
};
export const updateEntry = async ({ commit }, entry) => {
  const { id, date, picture, text } = entry;
  const updatedEntry = {
    date,
    picture,
    text,
  };
  const { data } = await journalApi.put(`/entries/${id}.json`, updatedEntry);
  commit("updateEntry", { id, ...data });
};
export const createEntry = async ({ commit }, entry) => {
  const { date, text, picture } = entry;
  const newEntry = {
    date,
    text,
    picture,
  };
  const { data } = await journalApi.post(`/entries.json`, newEntry);
  newEntry.id = data.name;
  commit("addEntry", newEntry);
  return newEntry.id;
};
export const deleteEntry = async ({ commit }, id) => {
  try {
    await journalApi.delete(`/entries/${id}.json`);
    commit("deleteEntry", id);
    return true;
  } catch (error) {
    console.log(error);
    return false;
  }
};
