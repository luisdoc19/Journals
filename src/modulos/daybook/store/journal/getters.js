export const getEntriesByTerm =
  (state) =>
  (term = "") => {
    if (term.length === 0) return state.entries;
    return state.entries.filter((entry) =>
      entry.text.toLowerCase().includes(term.toLowerCase())
    );
  };

export const getEntryById = (state) => (val) => {
  const entry = state.entries.find((entry) => entry.id === val);
  if (!entry) return;
  return { ...entry };
};
