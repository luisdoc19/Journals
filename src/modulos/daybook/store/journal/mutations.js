export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries];
  state.isLoading = false;
};
export const updateEntry = (state, entry) => {
  const index = state.entries.findIndex((e) => e.id === entry.id);
  state.entries[index] = entry;
};
export const addEntry = (state, entry) => {
  state.entries = [...state.entries, entry];
};
export const deleteEntry = (state, id) => {
  state.entries = state.entries.filter((e) => e.id !== id);
};

export const clearEntries = (state) => {
  state.entries = [];
};
