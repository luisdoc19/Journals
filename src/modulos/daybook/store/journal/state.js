export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: "Lorem ipsum dolor, sit amet consectetur adipisicing elit. Eligendi rerum quos veniam debitis iure quae dignissimos ex nobis aspernatur in numquam id cumque inventore sunt, necessitatibus odit delectus cupiditate maxime!",
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Saepe neque minus quam sequi. Nesciunt vero aliquid non quaerat doloremque nisi fugit aperiam velit blanditiis atque? Aut expedita repellendus labore quo?",
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: "Lorem, ipsum dolor sit amet consectetur adipisicing elit. In, est dignissimos laudantium harum, repellendus perferendis at labore, nihil maxime quo sint perspiciatis ab ullam deleniti dolorem illum asperiores voluptatem veniam.",
      picture: null,
    },
  ],
});
