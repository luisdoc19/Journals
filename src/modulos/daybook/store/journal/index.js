import state from "./state";
import * as getters from "./getters";
import * as mutations from "./mutations";
import * as actions from "./actions";

const myCustomModule = {
  namespaced: true,
  state,
  getters,
  mutations,
  actions,
};

export default myCustomModule;
