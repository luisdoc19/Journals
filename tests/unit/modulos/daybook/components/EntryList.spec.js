import { createStore } from "vuex";
import EntryListVue from "@/modulos/daybook/components/EntryList.vue";
import { shallowMount } from "@vue/test-utils";

import journal from "@/modulos/daybook/store/journal";
import { jornalState } from "../../../mock-data/test-jornal";

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

describe("Pruebas en el EntryList", () => {
  const store = createVuexStore(jornalState);
  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryListVue, {
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  it("debe llamar el getEntriesByTerm sin termino y mostrar 2 entradas", () => {
    expect(wrapper.findAll("entry-stub").length).toBe(2);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("debe de llamar getEntriesByTerm y filtrar resultados", async () => {
    const input = wrapper.find("input");
    await input.setValue("Mundo");

    expect(wrapper.findAll("entry-stub").length).toBe(1);
  });

  it("el boton de nuevo debe redireccionar a new", async () => {
    const btn = wrapper.find("button");
    await btn.trigger("click");
    expect(mockRouter.push).toHaveBeenCalled();
  });
});
