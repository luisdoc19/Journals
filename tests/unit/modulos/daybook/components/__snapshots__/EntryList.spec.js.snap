// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryList debe llamar el getEntriesByTerm sin termino y mostrar 2 entradas 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar entradas"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub data="[object Object]"></entry-stub>
    <entry-stub data="[object Object]"></entry-stub>
  </div>
</div>
`;
